src/utilities/identities/identities.ts

```diff
@@ -126,65 +132,46 @@ interface IdentityDidCrypto {
   ) => Promise<IEncryptedMessage>;
 }
 
-function extractSecretKey(keypair: KeyringPair) {
-  let secretKey: Uint8Array | undefined;
+function deriveAuthenticationKey(seed: Uint8Array): KeyringPair {
+  return getKeypairBySeed(seed).derive('//did//0');
+}
 
-  const { convertSecretKey } = ed2curve;
-  ed2curve.convertSecretKey = (secret) => {
-    secretKey = secret;
-    return convertSecretKey(secret);
+export function deriveEncryptionKeyFromSeed(seed: Uint8Array): {
+  type: string;
+  publicKey: Uint8Array;
+  secretKey: Uint8Array;
+} {
+  const edKeypair = sr25519PairFromSeed(seed);
```

Comment:
Just to make sure, This is `sr25519`, because the secret we extracted before was also sr25519, right?

Reply:
Yes, because all the keypairs in Sporran are sr25519. Actually I copied the code from SKYC at first, which was using ed25519. Luckily I have tried it with an existing identity+lightDID, created with the previous version of code, and the DID signing didn’t work, messages could not be decrypted. After I have updated the code to use the sr25519 the problem went away. Your comment highlighted to me that the name of the variable doesn’t match, so I have pushed a commit to address that.